Bundle/app/src/main/java/com/scottlindley/Bundle/Services/WeatherService.java

```diff
@@ -0,0 +1,176 @@
+            return;
+
+        //Build retrofit request
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(WEATHER_BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+
+        OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
+        Call<GsonCurrentWeather> call = service.getWeatherByLongLat(latitude, longitude, API_KEY, IMPERIAL_UNITS);
+        call.enqueue(new Callback<GsonCurrentWeather>() {
+            @Override
+            public void onResponse(Call<GsonCurrentWeather> call, Response<GsonCurrentWeather> response) {
+                if(response.isSuccessful()) {
+                    //see helper method
+                    handleResponse(response);
+                    jobFinished(jobParameters, false);
+                }else {
+                }
+            }
+
+            @Override
+            public void onFailure(Call<GsonCurrentWeather> call, Throwable t) {
+                t.printStackTrace();
+                //Send a name of 'failure' that will trigger the default case in the BroadcastReceiver
+                Intent intent = new Intent("service intent");
+                intent.putExtra("service name", "failure");
+
+                LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(intent);
+                jobFinished(jobParameters, false);
+            }
+        });
+    }
+
+    public void handleResponse(Response<GsonCurrentWeather> response){
+        /*Regardless of the api call (zip or longitude/latitude), the pieces we need in the JSON respose
+        are identical. Therefore we can use this block of code for both responses.
+        */
+
+        GsonCurrentWeather gsonWeather = response.body();
+        String cityName = gsonWeather.getName();
+        String description = gsonWeather.getWeather().get(0).getDescription();
+        String temperature = gsonWeather.getMain().getTemp();
+
+        showPersistentWeatherNotification(cityName, temperature);
+
+        Intent intent = new Intent("weather service");
+        intent.putExtra("city name", cityName);
+        intent.putExtra("description", description);
+        intent.putExtra("temperature", temperature);
+
+        LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(intent);
+
+    }
+
+    public void showPersistentWeatherNotification(String cityName, String temperature) {
+        String cityInNotification = "Weather for "+cityName+":";
+        String currentTempNotification = "Current temperature: "+ temperature +"\u2109";
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(getApplicationContext());
+        builder.setPriority(NotificationCompat.PRIORITY_MAX);
+        builder.setContentTitle(cityInNotification);
+        builder.setContentInfo(currentTempNotification);
+        builder.setSmallIcon(R.drawable.ic_wb_cloudy_black_24dp);
+        builder.setOngoing(true);
```

Comment:
I really suggest having the notification bring the user to the "home screen" when clicked. Otherwise the notification feels very disconnected from the rest of the app.